src/components/FullDetailView/FullDetailPage.js

```diff
@@ -57,71 +67,118 @@ export class FullDetailPage extends Component {
     const spec = { description: description, patches }
     curateAction(token, spec)
   }
 
+  // Action that calls the remote API that return a preview of the definition
+  previewDefinition(nextComponent) {
+    const { token, component, uiCurateGetDefinitionPreview } = this.props
+    const { changes } = this.state
+    if (!component && !nextComponent) return false
+
+    const previewComponent = nextComponent ? nextComponent : component
+    const patches = Contribution.buildPatch([], previewComponent, changes)
+    uiCurateGetDefinitionPreview(token, previewComponent, patches)
+  }
+
   // Shows the Modal to save a Contribution
   doPromptContribute() {
-    const { component } = this.props
-    if (!Contribution.hasChange(component)) return
+    const { changes } = this.state
+    if (isEmpty(changes)) return
     this.refs.contributeModal.open()
   }
 
-  handleSave(changes) {
-    const { onSave, component } = this.props
+  handleSave() {
+    const { onSave, component, uiCurateResetDefinitionPreview } = this.props
+    const { changes } = this.state
     const newComponent = { ...component, changes }
-    onSave && onSave(component, newComponent)
+    this.setState({ changes: {} }, () => {
+      uiCurateResetDefinitionPreview()
+      onSave && onSave(component, newComponent)
+    })
   }
 
   handleClose() {
     const { onClose } = this.props
-    onClose()
-    this.setState({ visible: false })
-  }
-
-  // Function called when a data has been changed
-  onChange(item, value) {
-    const { component } = this.props
     const { changes } = this.state
-    this.setState({ changes: Contribution.onChange(component, changes, item, value) }, () =>
-      console.log(this.state.changes)
+    if (isEmpty(changes)) return onClose()
+    const key = `open${Date.now()}`
+    const btn = (
+      <Fragment>
+        <AntdButton
+          type="primary"
+          size="small"
+          onClick={() => {
+            this.close()
+            notification.close(key)
+          }}
+        >
+          Confirm
+        </AntdButton>
+        <AntdButton type="secondary" size="small" onClick={() => notification.close(key)}>
+          Dismiss Notification
+        </AntdButton>
+      </Fragment>
     )
+    notification.open({
+      message: 'Unsaved Changes',
+      description:
+        'Some information have been changed and are currently unsaved. Are you sure to continue without saving?',
+      btn,
+      key,
+      onClose: notification.close(key),
+      duration: 0
+    })
   }
 
-  getValue(field) {
-    const { component } = this.props
-    const { changes } = this.state
-    return Contribution.getValue(component, changes, field)
+  close() {
+    const { uiCurateResetDefinitionPreview, onClose } = this.props
+    this.setState({ changes: {} }, () => {
+      uiCurateResetDefinitionPreview()
+      onClose()
+    })
   }
 
-  classIfDifferent(field, className) {
+  // Function called when a data has been changed
```

Comment:
`onChange` means exactly what that comment says so the comment is redundant. I'm all for comments where they add value. otherwise they are just some more code that needs to be maintained. So many times I've been led astray by "helpful" comments that in the end were wrong because the code changed and the comment didn't.